widgets/views/adframe.php

```diff
@@ -1,37 +1,49 @@
 <?php
 
 use yii\helpers\Url;
-use humhub\models\Setting;
-use humhub\modules\adsense\Assets;
+use humhub\libs\Html;
+use humhub\widgets\PanelMenu;
+use humhub\modules\ui\view\components\View;
 
-Assets::register($this);
-?>
+/* @var $client string */
+/* @var $slot string */
+/* @var $this View */
 
-<div class="panel">
-  <div class="panel-heading">
-    <?=Yii::t('AdsenseModule.base', '<strong>Community</strong> Ad'); ?>
-  </div>
-  <div class="panel-body">
-  <?php
-  	if (!Setting::Get('client', 'adsense')) {
- ?>
- 	<p>Please set your ad client and ad slot ids in administration</p>
- <?php
-	} else {
-?>
-<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
-<!-- In Post Correct -->
-	<ins class="adsbygoogle"
-     style="display:block"
-     data-ad-client="<?php print Setting::Get('client', 'adsense');?>"
-     data-ad-slot="<?php print Setting::Get('slot', 'adsense');?>"
-     data-ad-format="auto"></ins>
-<script>
-(adsbygoogle = window.adsbygoogle || []).push({});
-</script>
-<?php
-	}
+$this->registerJsFile('https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js', ['data-ad-client' => Yii::$app->getModule('adsense')->getClient(), 'async' => 'async', 'position' => View::POS_HEAD]);
```

Comment:
Not exactly sure if this is the right or wrong way of implementing, but seems to be what works?

See https://github.com/humhub/humhub/issues/4802#issuecomment-769059886